python/ray/serve/tests/test_deploy.py

```diff
@@ -833,15 +833,6 @@ class BadNumReplicas:
     with pytest.raises(ValidationError):
         Base.options(num_replicas="hi")
 
-    with pytest.raises(ValidationError):
-
-        @serve.deployment(num_replicas=0)
-        class ZeroNumReplicas:
-            pass
-
-    with pytest.raises(ValidationError):
-        Base.options(num_replicas=0)
-
```

Comment:
hmm `num_replicas=0` should still be invalid at the API level without autoscaling.